File: P2/ejercicio3/lib/guerrero_builder.dart
import 'package:ejercicio3/armadura_simple.dart';
import 'package:ejercicio3/personaje_builder.dart';
import 'package:ejercicio3/armadura.dart';

class GuerreroBuilder extends PersonajeBuilder {
  @override
  void addArmadura() {
    super.personaje?.armadura = ArmaduraSimple("Armadura Básica");
  }

  @override
  void addHabilidad() {
    super.personaje?.habilidad = "Lucha";
  }

  @override
  void addArma() {
    super.personaje?.arma = "Lanzallamas";
  }

  @override
  void setArmadura(Armadura armadura) {
    super.personaje?.armadura = armadura;
  }

  @override
  Armadura? getArmadura() {
    return super.personaje?.armadura;
  }
}

File: P1/P1/Ejercicio3/GuerreroBuilder.py
from PersonajeBuilder import PersonajeBuilder
from Armadura import Armadura
from ArmaduraSimple import ArmaduraSimple
from FuegoDecorador import FuegoDecorador

class GuerreroBuilder(PersonajeBuilder):
    def add_arma(self):
        self.personaje.arma = "Lanzallamas"

    def add_armadura(self):
        self.personaje.armadura = ArmaduraSimple("Armadura Basica")

    def add_habilidad(self):
        self.personaje.habilidad = "Lucha"
    
    def set_armadura(self, armadura):
        self.personaje.armadura = armadura
        
    def get_armadura(self):
        return self.personaje.armadura